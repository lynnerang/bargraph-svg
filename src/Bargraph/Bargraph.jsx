import * as React from 'react';
import { SvgLoader, SvgProxy } from 'react-svgmt';
import Image from '../images/o9-Bargraph-01.svg';
import PropTypes from 'prop-types';

const Bargraph = props => {
	const { volts } = props;
	const voltsWhole = Math.floor(volts);

	return (
		<SvgLoader path={Image}>
			<SvgProxy selector={`.diodes .st0:nth-child(-n+${voltsWhole})`} class="st0 active" />
			<SvgProxy selector={`.diodes .st0:nth-child(n+${voltsWhole + 1})`} class="st0" />
			<SvgProxy selector={`.diodes .st0:nth-child(${volts - voltsWhole ? voltsWhole + 1 : 0})`} class="st0 partial" />
		</SvgLoader>
	);
};

Bargraph.propTypes = {
	volts: PropTypes.number
};

export default Bargraph;
