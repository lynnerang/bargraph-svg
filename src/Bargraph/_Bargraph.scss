svg {
  width: 300px;
  height: 200px;
}

.active {
  fill: rgb(88, 223, 88)!important;
}

.partial {
  fill: rgb(255, 180, 67)!important;
}

@media (max-width: 420px) { 
  svg {
    width: 250px;
    height: 175px;
    margin-top: -3px;
  }
}