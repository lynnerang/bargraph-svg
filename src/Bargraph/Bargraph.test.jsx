import React from 'react';
import ReactDOM from 'react-dom';
import Bargraph from './Bargraph';
import { mount, render } from 'enzyme';

describe('Bargraph', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Bargraph />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot with volts of 0', () => {
    const wrapper = mount(<Bargraph volts={0} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with volts of 1', () => {
    const wrapper = mount(<Bargraph volts={1} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with volts of 2', () => {
    const wrapper = mount(<Bargraph volts={2} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with volts of 3', () => {
    const wrapper = mount(<Bargraph volts={3} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with volts of 4', () => {
    const wrapper = mount(<Bargraph volts={4} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with volts of 5', () => {
    const wrapper = mount(<Bargraph volts={5} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot when volts is not a whole integer', () => {
    const wrapper = mount(<Bargraph volts={4.532} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('adds a class of "active" to lit diodes based on total volts received', () => {
    const wrapper = mount(<Bargraph volts={3} />);
    setTimeout(() => expect(wrapper.find('.active').length).toEqual(3), 0);
  })

  it('adds a class of "partial" to the next diode if between 0 to 1000 milliVolts are received', () => {
    const wrapper = mount(<Bargraph volts={3.678} />);
    setTimeout(() => expect(wrapper.find('.active').length).toEqual(3), 0);
    setTimeout(() => expect(wrapper.find('.partial').length).toEqual(1), 0);
  })

  it('does not add any class of "partial" if the volts received is a whole integer', () => {
    const wrapper = mount(<Bargraph volts={4} />);
    setTimeout(() => expect(wrapper.find('.active').length).toEqual(4), 0);
    setTimeout(() => expect(wrapper.find('.partial').length).toEqual(0), 0);
  })
});