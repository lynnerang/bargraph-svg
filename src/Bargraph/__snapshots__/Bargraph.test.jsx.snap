// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bargraph matches the snapshot when volts is not a whole integer 1`] = `
<Bargraph
  volts={4.532}
>
  <SvgLoader
    onSVGReady={[Function]}
    path="o9-Bargraph-01.svg"
    style={null}
    svgXML={null}
  >
    <ReactSVG
      callback={[Function]}
      path="o9-Bargraph-01.svg"
      style={null}
      svgXML={null}
    >
      <svg
        data-src="o9-Bargraph-01.svg"
        style={null}
      />
    </ReactSVG>
  </SvgLoader>
</Bargraph>
`;

exports[`Bargraph matches the snapshot with volts of 0 1`] = `
<Bargraph
  volts={0}
>
  <SvgLoader
    onSVGReady={[Function]}
    path="o9-Bargraph-01.svg"
    style={null}
    svgXML={null}
  >
    <ReactSVG
      callback={[Function]}
      path="o9-Bargraph-01.svg"
      style={null}
      svgXML={null}
    >
      <svg
        data-src="o9-Bargraph-01.svg"
        style={null}
      />
    </ReactSVG>
  </SvgLoader>
</Bargraph>
`;

exports[`Bargraph matches the snapshot with volts of 1 1`] = `
<Bargraph
  volts={1}
>
  <SvgLoader
    onSVGReady={[Function]}
    path="o9-Bargraph-01.svg"
    style={null}
    svgXML={null}
  >
    <ReactSVG
      callback={[Function]}
      path="o9-Bargraph-01.svg"
      style={null}
      svgXML={null}
    >
      <svg
        data-src="o9-Bargraph-01.svg"
        style={null}
      />
    </ReactSVG>
  </SvgLoader>
</Bargraph>
`;

exports[`Bargraph matches the snapshot with volts of 2 1`] = `
<Bargraph
  volts={2}
>
  <SvgLoader
    onSVGReady={[Function]}
    path="o9-Bargraph-01.svg"
    style={null}
    svgXML={null}
  >
    <ReactSVG
      callback={[Function]}
      path="o9-Bargraph-01.svg"
      style={null}
      svgXML={null}
    >
      <svg
        data-src="o9-Bargraph-01.svg"
        style={null}
      />
    </ReactSVG>
  </SvgLoader>
</Bargraph>
`;

exports[`Bargraph matches the snapshot with volts of 3 1`] = `
<Bargraph
  volts={3}
>
  <SvgLoader
    onSVGReady={[Function]}
    path="o9-Bargraph-01.svg"
    style={null}
    svgXML={null}
  >
    <ReactSVG
      callback={[Function]}
      path="o9-Bargraph-01.svg"
      style={null}
      svgXML={null}
    >
      <svg
        data-src="o9-Bargraph-01.svg"
        style={null}
      />
    </ReactSVG>
  </SvgLoader>
</Bargraph>
`;

exports[`Bargraph matches the snapshot with volts of 4 1`] = `
<Bargraph
  volts={4}
>
  <SvgLoader
    onSVGReady={[Function]}
    path="o9-Bargraph-01.svg"
    style={null}
    svgXML={null}
  >
    <ReactSVG
      callback={[Function]}
      path="o9-Bargraph-01.svg"
      style={null}
      svgXML={null}
    >
      <svg
        data-src="o9-Bargraph-01.svg"
        style={null}
      />
    </ReactSVG>
  </SvgLoader>
</Bargraph>
`;

exports[`Bargraph matches the snapshot with volts of 5 1`] = `
<Bargraph
  volts={5}
>
  <SvgLoader
    onSVGReady={[Function]}
    path="o9-Bargraph-01.svg"
    style={null}
    svgXML={null}
  >
    <ReactSVG
      callback={[Function]}
      path="o9-Bargraph-01.svg"
      style={null}
      svgXML={null}
    >
      <svg
        data-src="o9-Bargraph-01.svg"
        style={null}
      />
    </ReactSVG>
  </SvgLoader>
</Bargraph>
`;
